ckanext/feedback/command/feedback.py

```diff
@@ -68,198 +82,84 @@ def get_connection(host, port, dbname, user, password):
-            'Request', 'Question', 'Advertise', 'Thank'
-            );
-        CREATE TABLE resource_comment (
-            id TEXT NOT NULL,
-            resource_id TEXT NOT NULL,
-            category resource_comment_category NOT NULL,
-            content TEXT,
-            rating INTEGER,
-            created TIMESTAMP,
-            approval BOOLEAN DEFAULT false,
-            approved TIMESTAMP,
-            approval_user_id TEXT,
-            PRIMARY KEY (id),
-            FOREIGN KEY (resource_id) REFERENCES resource (id),
-            FOREIGN KEY (approval_user_id) REFERENCES public.user (id)
-        );
-
-         CREATE TABLE resource_comment_reply (
-            id TEXT NOT NULL,
-            resource_comment_id TEXT NOT NULL,
-            content TEXT,
-            created TIMESTAMP,
-            creator_user_id TEXT,
-            PRIMARY KEY (id),
-            FOREIGN KEY (resource_comment_id) REFERENCES resource_comment (id),
-            FOREIGN KEY (creator_user_id) REFERENCES public.user (id)
-        );
-
-         CREATE TABLE resource_comment_summary (
-            id TEXT NOT NULL,
-            resource_id TEXT NOT NULL,
-            comment INTEGER,
-            rating NUMERIC,
-            created TIMESTAMP,
-            updated TIMESTAMP,
-            PRIMARY KEY (id),
-            FOREIGN KEY (resource_id) REFERENCES resource (id)
-        );
-    '''
-    )
-
+    engine = get_engine(host, port, dbname, user, password)
+    try:
+        if not modules:
+            drop_utilization_tables(engine)
+            create_utilization_tables(engine)
+            drop_resource_tables(engine)
+            create_resource_tables(engine)
+            drop_download_tables(engine)
+            create_download_tables(engine)
+            click.secho('Initialize all modules: SUCCESS', fg='green', bold=True)
+        elif 'utilization' in modules:
+            drop_utilization_tables(engine)
+            create_utilization_tables(engine)
+            click.secho('Initialize utilization: SUCCESS', fg='green', bold=True)
+        elif 'resource' in modules:
+            drop_resource_tables(engine)
+            create_resource_tables(engine)
+            click.secho('Initialize resource: SUCCESS', fg='green', bold=True)
+        elif 'download' in modules:
+            drop_download_tables(engine)
+            create_download_tables(engine)
+            click.secho('Initialize download: SUCCESS', fg='green', bold=True)
+    except Exception as e:
+        toolkit.error_shout(e)
+        sys.exit(1)
 
-def _create_download_tables(cursor):
-    cursor.execute(
-        '''
-        CREATE TABLE download_summary (
-            id TEXT NOT NULL,
-            resource_id TEXT NOT NULL,
-            download INTEGER,
-            created TIMESTAMP,
-            updated TIMESTAMP,
-            PRIMARY KEY (id),
-            FOREIGN KEY (resource_id) REFERENCES resource (id)
-        );
-    '''
-    )
+    engine.dispose()
```

Comment:
`Engine#dispose`をあまり明示的に呼ぶことは無いように思うのですが、どこか参考にしたコードなどありますでしょうか？

[公式マニュアル](https://docs.sqlalchemy.org/en/13/core/connections.html#engine-disposal)によると、`Engine#dispose`を呼ぶケースとして3パターンほど書かれていますが、だいぶ限定されたケースのように思われます。

> Valid use cases for calling [Engine.dispose()](https://docs.sqlalchemy.org/en/13/core/connections.html#sqlalchemy.engine.Engine.dispose) include:
> 
> - When a program wants to release any remaining checked-in connections held by the connection pool and expects to no longer be connected to that database at all for any future operations.
> 
> - When a program uses multiprocessing or fork(), and an [Engine](https://docs.sqlalchemy.org/en/13/core/connections.html#sqlalchemy.engine.Engine) object is copied to the child process, [Engine.dispose()](https://docs.sqlalchemy.org/en/13/core/connections.html#sqlalchemy.engine.Engine.dispose) should be called so that the engine creates brand new database connections local to that fork. Database connections generally do not travel across process boundaries.
> 
> - Within test suites or multitenancy scenarios where many ad-hoc, short-lived [Engine](https://docs.sqlalchemy.org/en/13/core/connections.html#sqlalchemy.engine.Engine) objects may be created and disposed.

Reply:
ご指摘頂き、ありがとうございます。

SQLAlchemyへ書き換える前に利用していた```session```や```cursor```等と同様に閉じる必要があるのかと考え、```Engine#dispose```を呼び出しておりました。

教えて頂いた公式のドキュメントより、今回は```Engine#dispose```を呼び出す必要はないため、該当箇所を削除いたしました。